modules/modal/react/lib/Modal.tsx

```diff
@@ -10,12 +11,36 @@ export enum ModalWidth {
 }
 
 export interface ModalProps extends React.HTMLAttributes<HTMLDivElement> {
+  /**
+   * This will add a `data-testid` attribute to the Popup component, not the container
+   */
+  testId?: string;
```

Comment:
Oh, yes. I was going to propose this as an official optional property with a helper function that includes for testing. I've done this in the past and it worked well. Some teams use `cy-test`, others use `data-testid`, others use `data-test-id` and I think I've seen `data-automation-id`.

Since our components mostly just mix extra attributes into the root component node, this isn't strictly needed and perhaps should not be introduced here.

Reply:
Ah. So the `Container` mixes in `elemProps` and this adds the testid to the `[role=dialog]` element. The test ID could be added to the container and the modal helper instead tries to find the test id and then returns the modal element.

Reply:
I wanted to explicitly call out the relationship between the `testId` in the code and the `testId` in the helper rather than it being an implicit relationship.

@lychyi @anicholls What do you think?

Reply:
I'm indifferent, if this isn't a documented prop, it can really change at anytime.

Reply:
Personally, I'd prefer having a slightly more brittle test (using `elemProps`) than adding undocumented props that aren't meant to be used in prod, but I'm not too torn either way.

Reply:
I'll remove and this and we can make an official proposal later. This would be something that all components would have as a documented optional prop that would be used in production with the option to enable/disable.

In the platform, automation IDs are present even in the production HTML code. I think it is preferable to opt-in during the testing phase, but I would be fine if they are always present like we currently have in the product.